api/grpcserver/globalstate_service.go

```diff
@@ -448,18 +496,54 @@ func (s GlobalStateService) GlobalStateStream(in *pb.GlobalStateStreamRequest, s
 		channelLayer   chan events.NewLayer
 	)
 	if filterAccount {
-		channelAccount = events.GetAccountChannel()
+		if s.globalStateStreamAccountChannel == nil {
+			channelAccount = events.SubscribeToAccounts()
+			if channelAccount == nil {
+				log.Error("Reporter is not running")
+				return errors.New("Reporter is not running")
+			}
+		} else {
+			channelAccount = s.globalStateStreamAccountChannel
+		}
 	}
 	if filterReward {
-		channelReward = events.GetRewardChannel()
+		// needs to be buffered to avoid reporter from being blocked on slow streaming connections
+		if s.globalStateStreamRewardsChannel == nil {
+			channelReward = events.SubscribeToRewards()
+			if channelReward == nil {
+				log.Error("Reporter is not running")
+				return errors.New("Reporter is not running")
+			}
+		} else {
+			channelReward = s.globalStateStreamRewardsChannel
+		}
+		if channelReward == nil {
+			// what to do here
```

Comment:
Do you need this check? None of the other channel logic has it. Seems redundant/unnecessary.